src/common/io/socket/common.c

```diff
@@ -129,6 +141,52 @@ sckOptionSet(int fd)
     FUNCTION_TEST_RETURN_VOID();
 }
 
+/**********************************************************************************************************************************/
+static bool
+sckConnectInProgress(int errNo)
+{
+    return errNo == EINPROGRESS || errNo == EINTR;
+}
+
+void
+sckConnect(int fd, const String *host, unsigned int port, const struct addrinfo *hostAddress, TimeMSec timeout)
+{
+    FUNCTION_LOG_BEGIN(logLevelTrace);
+        FUNCTION_LOG_PARAM(INT, fd);
+        FUNCTION_LOG_PARAM(STRING, host);
+        FUNCTION_LOG_PARAM(UINT, port);
+        FUNCTION_LOG_PARAM_P(VOID, hostAddress);
+        FUNCTION_LOG_PARAM(TIME_MSEC, timeout);
+    FUNCTION_LOG_END();
+
```

Comment:
Are we missing some ASSERTs here - can *host, or *host Address be NULL?

Reply:
Yes, will add.